src/scheduler/listDagRuns.tsx

```diff
@@ -0,0 +1,243 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { useTable, useGlobalFilter, usePagination } from 'react-table';
+import { ClipLoader } from 'react-spinners';
+import TableData from '../utils/tableData';
+import { ICellProps } from '../utils/utils';
+import { SchedulerService } from './schedulerServices';
+import { Dayjs } from 'dayjs';
+
+const ListDagRuns = ({
+  composerName,
+  dagId,
+  startDate,
+  endDate,
+  setDagRunId,
+  selectedDate,
+
+  setBlueListDates,
+  setGreyListDates,
+  setOrangeListDates,
+  setRedListDates,
+  setGreenListDates,
+  setDarkGreenListDates
+}: {
+  composerName: string;
+  dagId: string;
+  startDate: string;
+  endDate: string;
+  setDagRunId: (value: string) => void;
+  selectedDate: Dayjs | null;
+
+  setBlueListDates: (value: string[]) => void;
+  setGreyListDates: (value: string[]) => void;
+  setOrangeListDates: (value: string[]) => void;
+  setRedListDates: (value: string[]) => void;
+  setGreenListDates: (value: string[]) => void;
+  setDarkGreenListDates: (value: string[]) => void;
+}): JSX.Element => {
+  const [dagRunsList, setDagRunsList] = useState([]);
+  const [dagRunsCurrentDateList, setDagRunsCurrentDateList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const data =
+    dagRunsCurrentDateList.length > 0 ? dagRunsCurrentDateList : dagRunsList;
+  const columns = React.useMemo(
```

Comment:
don't need to use useMemo here.

Reply:
Here it is required to ensure that the 'columns' array is only recomputed when its dependencies change. Needed for React-table